Bot/Services/DatabaseHandler.cs

```diff
@@ -0,0 +1,113 @@
+                            .WithColor(Color.Red)
+                            .Build();
+                        await command.RespondAsync(embed: embed, ephemeral: true);
+                        return false;
+                    }
+                    else return true;
+                } 
+                catch (Exception ex)
+                {
+                    log.Error(ex.Message, ex);
+                    return true;
+                }
+            }
+        }
+
+        public static async Task CheckCreateUser(ulong id)
+        {
+            string sql = "SELECT * FROM users WHERE id = :userID";
+
+            using (OracleConnection connection = new(connectionString))
+            using (OracleCommand command = new(sql, connection))
+            {
+                await connection.OpenAsync();
+                command.Parameters.Add(new OracleParameter("userID", (long)id));
+                var result = await command.ExecuteScalarAsync();
+                if (result is null)
+                {
+                    OracleCommand createUser = new("INSERT INTO users (id) VALUES (:userID)", connection);
+                    createUser.Parameters.Add("userID", (long)id);
+                    await createUser.ExecuteNonQueryAsync();
+                }
+            }
+        }
+
+        public static async Task<(bool? previous, bool? result)> UpdateDNDTerms(ulong discordId, bool value)
+        {
+            int agreeToTerms = value ? 1 : 0;
+            short? pdbValue;
+            short? dbValue;
+
+            string sql = "UPDATE users SET AGREETODND = :agree WHERE id = :id";
+
+            try
+            {
+                await CheckCreateUser(discordId);
+
+                using (OracleConnection connection = new(connectionString))
+                using (OracleCommand command = new(sql, connection))
+                using (OracleCommand checkValue = new("SELECT agreeToDND FROM users WHERE id = :id", connection))
+                {
+                    await connection.OpenAsync();
+                    command.Parameters.Add(new OracleParameter("agree", agreeToTerms));
+                    command.Parameters.Add(new OracleParameter("id", (long)discordId));
+                    checkValue.Parameters.Add("id", (long)discordId);
+
+
+                    pdbValue = (short?)await checkValue.ExecuteScalarAsync();
+                    await command.ExecuteScalarAsync();
+                    dbValue = (short?)await checkValue.ExecuteScalarAsync();
+                }
+            }
+            catch (Exception ex)
+            {
+                log.Error(ex);
+                return (null, null);
+            }
+
+            bool previous = (pdbValue == 0) ? false : true;
```

Comment:
If the value was `bool` to begin with, ternary wouldn't be required here. This method reduces readability.

Reply:
Nevermind, it turns out Oracle just sucks.

Reply:
yeah lol